LayoutTests/http/tests/paymentrequest/paymentrequest-merchantCategoryCode.https.html

```diff
@@ -0,0 +1,58 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>Tests for providing `merchantCategoryCode` as part of `ApplePayRequestBase`.</title>
+<script src="/js-test-resources/ui-helper.js"></script>
+<script src="/resources/payment-request.js"></script>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<body>
+<script>
+    setup({ explicit_done: true, explicit_timeout: true });
+    
+    test((test) => {
+        const method = validPaymentMethod();
+        method.data.features = ["paymentRequestMerchantCategoryCode"];
+        new PaymentRequest([method], validPaymentDetails());
+    }, "Should have a feature for `paymentRequestMerchantCategoryCode`.");
+    
+    user_activation_test(async (test) => {
+        let request = new PaymentRequest([validPaymentMethod()], validPaymentDetails());
+        request.addEventListener("merchantvalidation", (event) => {
+            event.complete({ });
+        });
+        request.addEventListener("shippingaddresschange", (event) => {
+            assert_equals(internals.mockPaymentCoordinator.merchantCategoryCode, null, "check that the `merchantCategoryCode` is still `null` after an update");
+            event.updateWith({ });
+            internals.mockPaymentCoordinator.acceptPayment();
+        });
+        
+        let response = await request.show();
```

Comment:
I wonder if you should wrap this in a promise to assure sure it's actually getting run before the test finishes. I would do something like: 

```suggestion
        const shippingPromise = new Promise(resolve => {
            request.addEventListener("shippingaddresschange", (event) => {
                assert_equals(internals.mockPaymentCoordinator.merchantCategoryCode, null, "check that the `merchantCategoryCode` is still `null` after an update");
                event.updateWith({ });
                internals.mockPaymentCoordinator.acceptPayment();
                resolve();
            });
        });

        const response = await request.show();
        await shippingPromise;
```